cdap-api/src/main/java/co/cask/cdap/api/schedule/ConstraintProgramScheduleBuilder.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+/**
+ * Similar to {@link ScheduleBuilder}, but allows specifying whether the schedule should wait until the
+ * configured constraint is met.
+ */
+public interface ConstraintProgramScheduleBuilder extends ScheduleBuilder {
+
+  /**
+   * Specifies that the schedule should wait until the configured constraint is met.
+   */
+  ScheduleBuilder waitUntilMet();
```

Comment:
Edit: Updated TimeRangeConstraint to default to `waitUntilMet`, and added the method `abortIfNotMet` to the builder api.